### PR TITLE
Loads firebase secret from Taskcluster for nightly

### PR DESCRIPTION
**Pull request by mitchhentges**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features
- [x] Only inject Firebase credentials for `nightly` Fenix
- [x] [Staging task group has produced an APK with staging firebase credentials baked in](https://tools.taskcluster.net/groups/LJYH-equRW6svLGtiIEW1A)
- [x] [`ci-configuration` changes landed and applied](https://phabricator.services.mozilla.com/D37491)

### CODE REVIEW

File: automation/taskcluster/lib/tasks.py [hunk 1]
```diff
@@ -60,11 +60,12 @@ def craft_assemble_release_task(self, architectures, channel, is_staging, versio
             'python automation/taskcluster/helper/get-secret.py -s {} -k {} -f {}'.format(
                 secret_index, key, target_file
             )
-            for key, target_file in (
+            for key, target_file in filter(None.__ne__, (
```
Comment (mitchhentges) [inline automation/taskcluster/lib/tasks.py]: Is this readable enough? It's removing all `None`s from the list.
I could make this a list comprehension instead :thinking:

File: .taskcluster.yml [hunk 2]
```diff
@@ -78,7 +78,6 @@ tasks:
           retries: 5
           routes:
             - statuses  # Automatically added by taskcluster-github. It must be explicit because of Chain of Trust
-            - tc-treeherder.v2.fenix.${head_rev}
```
Comment (JohanLorenzo) [inline .taskcluster.yml]: Nit: Let's make this route conditional. I had started https://github.com/mozilla-mobile/fenix/pull/2921. I'm fine closing it so it's merged in this PR.

File: automation/taskcluster/decision_task.py [hunk 3]
```diff
@@ -211,9 +212,9 @@ def nightly_to_production_app(is_staging, version_name):
         beta_semver = re.compile(r'^v\d+\.\d+\.\d+-beta\.\d+$')
         production_semver = re.compile(r'^v\d+\.\d+\.\d+(-rc\.\d+)?$')
         if beta_semver.match(result.tag):
-            ordered_groups_of_tasks = release('beta', False, version)
+            ordered_groups_of_tasks = release('beta', result.staging, version)
```
Comment (JohanLorenzo) [inline automation/taskcluster/decision_task.py]: :+1:

File: .taskcluster.yml [hunk 4]
```diff
@@ -181,7 +180,7 @@ tasks:
                         git fetch ${repository} refs/tags/${head_rev}
                         && git config advice.detachedHead false
                         && git checkout FETCH_HEAD
-                        && python automation/taskcluster/decision_task.py github-release ${event.release.tag_name}
+                        && python automation/taskcluster/decision_task.py github-release ${event.release.tag_name} --staging
```
Comment (JohanLorenzo) [inline .taskcluster.yml]: Nit: Let's make this flag dynamic too

File: automation/taskcluster/lib/tasks.py [hunk 5]
```diff
@@ -374,9 +375,7 @@ def _craft_default_task_definition(
             "priority": self.tasks_priority,
             "dependencies": [self.task_id] + dependencies,
             "requires": "all-completed",
-            "routes": routes + [
-                "tc-treeherder.v2.fenix.{}".format(self.commit)
```
Comment (JohanLorenzo) [inline automation/taskcluster/lib/tasks.py]: Nit: Let's keep this route based on the trust level.

**Comment by mitchhentges** [comment]
Hey, sorry, this PR had staging-testing changes in it that wouldn't be in the final diff.
I'll re-assign you for review once I've peeled them out :)

File: .taskcluster.yml (hunk 2 above)
Comment (mitchhentges) [reply .taskcluster.yml]: Let's keep your PR, my route-removal was temporary

File: .taskcluster.yml (hunk 4 above)
Comment (mitchhentges) [reply .taskcluster.yml]: Removed this change from my PR, sorry for bad comms :)

**Comment by mitchhentges** [comment]
In the [staging task group](https://tools.taskcluster.net/groups/WqX-tjbIRy2XyTsD5x50Gg), I've modified the code to print out the `project_id` string (which is injected in a fake `firebase.xml` file). I've confirmed with `adb logcat | grep mhentges` that it shows up

**Comment by mitchhentges** [comment]
Fixes #2356

**Comment by mitchhentges** [comment]
Back to WIP, we want [`nightly` Firebase'd before `production`](https://github.com/mozilla-mobile/fenix/issues/2356#issuecomment-506064656)

**Comment by mitchhentges** [comment]
This has been updated to fetch secrets, regardless of release. I did a happy staging run [here](https://tools.taskcluster.net/groups/LJYH-equRW6svLGtiIEW1A).
Talking to @jonalmeida about landing this in 5 minutes in an in-person meeting

**Comment by mitchhentges** [comment]
I updated this PR so that, when "legacy nightly" builds are done (builds for the `nightly` track of `org.mozilla.fenix`), it gets the existing nightly secrets (e.g.: adjust/leanplum/sentry dsn), but gets the correct firebase credentials (for `org.mozilla.fenix`).
So, this PR now depends on [this landing and being applied](https://phabricator.services.mozilla.com/D37491)

**Pull request merged**